catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/ObjectAdapter.java

```diff
@@ -42,19 +48,29 @@ public static ObjectElement marshalFrom(Attribute attribute) throws CatalogTrans
               "Could not transform Metacard to XML.  Could not serialize Attribute Object value.",
               e);
         }
-        ((ObjectElement) element).getValue().add(baos.toByteArray());
+        (element).getValue().add(baos.toByteArray());
       }
     }
     return element;
   }
 
   public static Attribute unmarshalFrom(ObjectElement element) {
     AttributeImpl attribute = null;
-    for (Serializable value : element.getValue()) {
+    String elementName = element.getName();
+
+    for (byte[] value : element.getValue()) {
+      Serializable unmarshalledValue = value;
+      try (ObjectInputStream objectInputStream =
+          new ObjectInputStream(new ByteArrayInputStream(value))) {
+        unmarshalledValue = (Serializable) objectInputStream.readObject();
+      } catch (IOException | ClassNotFoundException | RuntimeException e) {
+        LOGGER.debug("Could not unmarshal Metacard object for attribute: " + elementName, e);
```

Comment:
✏️ Can we do this instead?
```suggestion
        LOGGER.debug("Could not unmarshal Metacard object for attribute: {}", elementName, e);
```